devito/ir/support/utils.py

```diff
@@ -69,11 +69,15 @@ def detect_flow_directions(exprs):
                 continue
             dimensions = [d for d in w.aindices if d is not None]
             for d in dimensions:
+                rootd = d
+                while rootd.is_Sub:
```

Comment:
it is safe to assume that subdimensions cannot be nested, otherwise we'll need a more general mechanism then this to retrieve the parent

so here you can just filter the dimensions list (before the ``for d in dimensions`` loop) as
```
dimensions = [d.parent if d.is_Sub else d for d in dimensions]
```
or alternatively stick an ``if d.is_Sub`` here

Reply:
Random thought: is Access.distance actually the right place to put this check?